heron/scheduler-core/src/java/com/twitter/heron/scheduler/RuntimeManagerMain.java

```diff
@@ -347,14 +347,18 @@ protected boolean validateRuntimeManage(
       return false;
     }
 
-    // Check whether cluster/role/environ matched
-    ExecutionEnvironment.ExecutionState executionState = adaptor.getExecutionState(topologyName);
-    if (executionState == null
-        || !executionState.getCluster().equals(Context.cluster(config))
-        || !executionState.getRole().equals(Context.role(config))
-        || !executionState.getEnviron().equals(Context.environ(config))) {
-      LOG.severe("cluster/role/environ not matched");
-      return false;
+    // Check whether cluster/role/environ matched. It is possible to get into a state where some of
+    // the data does not actually exist in zookeeper. If we are in this state them assume the
+    // runtime is valid.
+    if (adaptor.doesExecutionStateExist(topologyName)) {
```

Comment:
I think it is, if the execution state doesn't exist we cant properly validate the runtime so my proposal is that we default to true. If we return false the command will fail and we'd be in the same situation where the command will fail to run (see: https://github.com/twitter/heron/blob/master/heron/scheduler-core/src/java/com/twitter/heron/scheduler/RuntimeManagerMain.java#L309).

The manifestation of the bug is that the execution state got deleted from zookeeper, since that happens we cannot really do this validation step. Do you know what this validation was meant to protect against? It seems to be that its meant to make it so you cannot deploy a topology twice to two different cluster, role, environments correct? Can we always assume that the user does not want to do that?